pkg/virt-handler/device-manager/device_controller.go

```diff
@@ -275,15 +275,32 @@ func (c *DeviceController) RefreshMediatedDeviceTypes() {
 	}()
 }
 
+func (c *DeviceController) getExternallyProvidedMdevs() map[string]struct{} {
+	externalMdevResourcesMap := make(map[string]struct{})
+	if hostDevs := c.virtConfig.GetPermittedHostDevices(); hostDevs != nil {
+		if len(hostDevs.MediatedDevices) != 0 {
```

Comment:
you can skip this `if`, the for loop will just won't iterate over hostDevs.MediatedDevices when `len == 0`